rs/execution_environment/src/execution_environment.rs

```diff
@@ -3094,6 +3094,7 @@ impl ExecutionEnvironment {
                         Err(err.into())
                     }
                 };
+                canister.update_on_low_wasm_memory_hook_condition();
```

Comment:
This is the place that ensures we update the hook condition status after `install/upgrade/reinstall/uninstall` as desired.